kernel/src/syscall_driver.rs

```diff
@@ -179,9 +179,11 @@ impl From<process::Error> for CommandReturn {
 /// and what they represents) are specific to the peripheral system call
 /// driver.
 ///
-/// Note about subscribe: upcall subscriptions are handled entirely by the core
-/// kernel, and therefore there is no subscribe function for capsules to
-/// implement.
+/// Note about **subscribe** and **allow** syscalls:
+/// upcall subscriptions and buffer allows
+/// are handled entirely by the core kernel,
+/// and therefore there is no corresponding function
+/// for capsules to implement.
```

Comment:
Nitpick: these comments should be wrapped at 80 columns.

This isn't quite correct, because userspace-readable Allow calls do have a corresponding method. It's only Read-Only Allow and Read-Write Allow that have no corresponding method.

Reply:
Updated to mention userspace-readable allow.

If word wrapping in doc comments is relevant, I would suggest having it as part of the formatter run. It's a bother to reflow things manually. Thankfully, computers are good at this.

Reply:
> If word wrapping in doc comments is relevant, I would suggest having it as part of the formatter run. It's a bother to reflow things manually. Thankfully, computers are good at this.

My experience begs to differ :-p

Automatic formatters can easily detect a line that is too long, but when a line is wrapped too short, they can't tell whether the line was wrapped for alignment reasons (e.g. to make a table look good) or whether it should be unwrapped. Telling an autoformatter to automatically remove line wrapping results in lists and tables getting unwrapped into paragraphs.

As a result, `rustfmt` only supports automatically wrapping comments, not un-wrapping them.

Reply:
Rust standardizes doc comments as markdown, and markdown specifies paragraphs as their own thing, distinct from tables. It can be taken advantage of. It's a matter of taste, but I prefer manual adjustments to unusual constructs once in a while rather than having to manually re-break lines on every change in prose.